src/net/sourceforge/kolmafia/session/EquipmentManager.java

```diff
@@ -42,83 +49,110 @@
-  public static final int SLOTS = 11;
+    // Mutable equipment slots
+    HAT("hat"),
+    WEAPON("weapon"),
+    HOLSTER("holster"),
+    OFFHAND("off-hand", "offhand"),
+    CONTAINER("back", "container"),
+    SHIRT("shirt"),
+    PANTS("pants"),
+    ACCESSORY1("acc1"),
+    ACCESSORY2("acc2"),
+    ACCESSORY3("acc3"),
+    FAMILIAR("familiar", "familiarequip"),
 
-  // Pseudo-equipment slots
-  public static final int CROWNOFTHRONES = 11;
+    // Pseudo-equipment slots
+    CROWNOFTHRONES("crown-of-thrones", "crownofthrones"),
 
-  public static final int STICKER1 = 12;
-  public static final int STICKER2 = 13;
-  public static final int STICKER3 = 14;
+    STICKER1("sticker1", "st1"),
+    STICKER2("sticker2", "st2"),
+    STICKER3("sticker3", "st3"),
 
-  public static final int CARDSLEEVE = 15;
+    CARDSLEEVE("card-sleeve", "cardsleeve"),
 
-  public static final int FOLDER1 = 16;
-  public static final int FOLDER2 = 17;
-  public static final int FOLDER3 = 18;
-  public static final int FOLDER4 = 19;
-  public static final int FOLDER5 = 20;
+    FOLDER1("folder1"),
+    FOLDER2("folder2"),
+    FOLDER3("folder3"),
+    FOLDER4("folder4"),
+    FOLDER5("folder5"),
 
-  public static final int BUDDYBJORN = 21;
+    BUDDYBJORN("buddy-bjorn", "buddybjorn"),
 
-  public static final int BOOTSKIN = 22;
-  public static final int BOOTSPUR = 23;
+    BOOTSKIN("bootskin"),
+    BOOTSPUR("bootspur"),
 
-  // Count of all equipment slots: HAT to BOOTSPUR
-  public static final int ALL_SLOTS = 24;
+    FAKEHAND("fakehand");
 
-  public static final int FAKEHAND = 24;
+    public final String name;
+    public final String phpName;
+    private static final Slot[] VALUES = values();
+
+    Slot(String name) {
+      this.name = name;
+      this.phpName = name;
+    }
+
+    Slot(String name, String phpName) {
+      this.name = name;
+      this.phpName = phpName;
+    }
+
+    /** Slot names as they should appear when prompting for a slot input to a script */
+    public static final String[] NAMES =
+        Arrays.stream(VALUES).filter(x -> x != NONE).map(x -> x.name).toArray(String[]::new);
+
+    private static final Map<String, Slot> nameToSlot =
+        Arrays.stream(VALUES).collect(Collectors.toMap(type -> type.name, Function.identity()));
+
+    private static final Map<String, Slot> phpNameToSlot =
+        Arrays.stream(VALUES).collect(Collectors.toMap(type -> type.phpName, Function.identity()));
+
+    public static Slot byCaselessName(String name) {
+      return nameToSlot.getOrDefault(name.toLowerCase(), Slot.NONE);
+    }
+
+    public static Slot byCaselessPhpName(String name) {
+      return phpNameToSlot.getOrDefault(name.toLowerCase(), Slot.NONE);
+    }
+
+    public static Slot byOrdinal(int ordinal) {
+      return VALUES[ordinal];
+    }
+  }
+
+  // Count of slots visible on equipment page: HAT to FAMILIAR
+  public static final EnumSet<Slot> SLOTS = EnumSet.range(Slot.HAT, Slot.FAMILIAR);
```

Comment:
Can we give these EnumSets javadocs so the difference is clearer in an IDE? e.g. the difference between SLOTS and ALL_SLOTS isn't immediately obvious.